src/main/java/ch/njol/skript/expressions/ExprBlockDataTags.java

```diff
@@ -0,0 +1,68 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.bukkitutil.BlockDataUtils;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+@Name("Block Data Tags")
+@Description("Gets the block data tags associated with a blockdata.")
+@Examples("send block data tags of campfire")
+@RequiredPlugins("Minecraft 1.13+")
+@Since("INSERT VERSION")
+public class ExprBlockDataTags extends SimpleExpression<String> {
+
+	static {
+		PropertyExpression.register(ExprBlockDataTags.class, String.class, "[block[ ]data] tags", "blockdatas");
```

Comment:
block data shouldn't be optional, otherwise we might start to get confusion with material tags (#minecraft:logs), or with scoreboard tags.

Reply:
not resolved

Reply:
I've got changes on my local that I haven't pushed yet. I marked it as resolved as a note to myself